### PR TITLE
Fix scale problem during checkpoint loading for fsdp2

### PR DESCRIPTION
**Pull request by henrique**
### What does this PR do?

Only load HF checkpoints on rank 0 for fsdp2. Later the model is already broadcasted to all ranks in 'fsdp2_load_full_state_dict'

We have a problem when starting larger jobs, e.g. 512+ ranks, as each rank tries to simultaneously fetch the checkpoints from the file system/HF. This causes a massive resource contention on distributed file systems and greatly increases the length and variability of the loading, even at smaller scales. As we increase the number of ranks, it will eventually lead to nccl timeouts as the fastest rank has to wait for the slowest, which will break the initialization.

### Test

Model loading only happens on rank 0, is faster and doesn't timeout at scale; loss curve matches the original.

### CODE REVIEW

File: verl/workers/fsdp_workers.py [hunk 1]
```diff
@@ -366,12 +366,16 @@ def _build_model_optimizer(
                 else:
                     actor_module_class = AutoModel
 
-            actor_module = actor_module_class.from_pretrained(
-                pretrained_model_name_or_path=local_path,
-                torch_dtype=torch_dtype,
-                config=actor_model_config,
-                trust_remote_code=trust_remote_code,
-            )
+            if self.rank > 0 and self.config.actor.strategy == "fsdp2":
```
Comment (vermouth1992) [inline verl/workers/fsdp_workers.py:369]: I guess init_context should also include this feature?

**Comment by EduardDurech** [comment]
Related issue https://github.com/volcengine/verl/issues/2078

File: verl/workers/fsdp_workers.py (hunk 1 above)
Comment (henrique) [reply verl/workers/fsdp_workers.py:369]: Hi @vermouth1992 
I guess you mean something that can be used also by both CriticWorker and RewardModelWorker?

Comment (vermouth1992) [reply verl/workers/fsdp_workers.py:369]: Actually, we are deprecating FSDP/Megatron RewardModelWorker to favor server-based reward model worker

Comment (EduardDurech) [reply verl/workers/fsdp_workers.py:369]: @vermouth1992 what PR should we rebase?

**Comment by henrique** [comment]
closing in favor of https://github.com/volcengine/verl/pull/4644
